src/components/application_manager/test/commands/mobile/on_keyboard_input_notification_test.cc

```diff
@@ -0,0 +1,160 @@
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "application_manager/commands/mobile/on_keyboard_input_notification.h"
+#include "commands/commands_test.h"
+#include <application_manager/smart_object_keys.h>
+#include "application_manager/message.h"
+#include "application_manager/mock_message_helper.h"
+#include "command_impl.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+namespace strings = application_manager::strings;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+}
+
+using application_manager::commands::mobile::OnKeyBoardInputNotification;
+using application_manager::MockMessageHelper;
+using application_manager::commands::CommandImpl;
+using application_manager::ApplicationSet;
+using testing::Return;
+using testing::_;
+
+class OnKeyBoardInputNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  void SetSendNotificationExpectations(MessageSharedPtr msg) {
+    EXPECT_CALL(message_helper_, PrintSmartObject(_)).WillOnce(Return(false));
+    EXPECT_CALL(app_mngr_, SendMessageToMobile(msg, _));
+  }
+
+  void SetSendNotificationVariables(MessageSharedPtr msg) {
+    ASSERT_EQ(application_manager::MessageType::kNotification,
+              (*msg)[strings::params][strings::message_type].asInt());
+    ASSERT_EQ(CommandImpl::mobile_protocol_type_,
+              (*msg)[strings::params][strings::protocol_type].asInt());
+    ASSERT_EQ(CommandImpl::protocol_version_,
+              (*msg)[strings::params][strings::protocol_version].asInt());
+  }
+
+  MockAppPtr InitAppSetDataAccessor(SharedPtr<ApplicationSet>& app_set) {
+    app_set = (!app_set ? ::utils::MakeShared<ApplicationSet>() : app_set);
+    MockAppPtr app(CreateMockApp());
+    app_set->insert(app);
+    EXPECT_CALL(app_mngr_, applications())
+        .WillOnce(Return(DataAccessor<ApplicationSet>(*app_set, lock_)));
+    return app;
+  }
+
+  MockMessageHelper& message_helper_ =
+      *MockMessageHelper::message_helper_mock();
```

Comment:
@pvvasilev Please move this initialization in the constructor in the initializer list.